distributed/scheduler.py

```diff
@@ -1317,6 +1318,9 @@ def __init__(
         self.computations: deque[Computation] = deque(
             maxlen=dask.config.get("distributed.diagnostics.computations.max-history")
         )
+        self.erred_tasks: deque[TaskState] = deque(
+            maxlen=dask.config.get("distributed.diagnostics.erred-tasks.max-history")
+        )
```

Comment:
@crusaderky will keeping around a deque of references to old TaskStates that have failed cause any issues that you can think of?

Reply:
I would wipe out the run_spec, as it can be potentially several MB per tasks. It's what we already do with the `to_loggable` method in worker_state_machine.py.

Reply:
@ian-r-rose you should be able to do this by grepping for `del self.tasks` in scheduler.py.  Just before the del line we'd also `del `ts.run_spec`.

Reply:
I would also be fine just not tracking exceptions in this way and losing them when they go out of scope.  I'm not sure that the benefit added by this page outweighs the extra state in the scheduler.  My inclination would be to remove the deque entirely if that's possible.

Reply:
> I would also be fine just not tracking exceptions in this way and losing them when they go out of scope. I'm not sure that the benefit added by this page outweighs the extra state in the scheduler. My inclination would be to remove the deque entirely if that's possible.

It's definitely possible (and not hard), I am just not convinced that it results in a good user experience. To get really concrete, if I run the following:

```python
import time

import dask
import distributed

cluster = distributed.LocalCluster()
client = distributed.Client(cluster)

@dask.delayed
def func(x):
    time.sleep(1)
    raise ValueError(f"Oops got {x}")
    
func(10).compute()
```

The exceptions are not reachable on the dashboard, because there are no futures hanging on to the task reference, and they are very quickly cleaned up by the scheduler. This is true of basically any `.compute()` call that errors. In practice, I think that means that a *lot* of exceptions would not be visible, and it would not be possible to do any sort of forensics after-the-fact. Having a relatively short data structure that keeps them for a bit longer results in a real UX improvement.